### DISCUSSION TITLE
Transvector Dislocator Not Working

### DISCUSSION THREAD

**Discussion by DylanTaylor1** [DylanTaylor1/GTNH-CropAutomation#2] [Bugs]
If you want to report a bug or have a suggestion, leave it here!

**Comment by Grubasov** [comment]
Hi I have a little problem. My robot from time to time change position of Transvector Dislocator. He moves it on blank farmland between farms. How can i fix this problem?

![1](https://github.com/DylanTaylor1/ic2-crop-automation/assets/138994845/f68e3e33-55e5-40b0-ac7e-b564616f21e8)
![2](https://github.com/DylanTaylor1/ic2-crop-automation/assets/138994845/5011c7d6-93f2-4fd5-b9e4-c77fd889e383)

**Reply by DylanTaylor1** [reply]
@Grubasov Hi there, thanks for sharing!

ISSUE: The Transvector Dislocator teleports instead of the crop.

WHY THIS HAPPENS: The order of the transvector binder was somehow messed up (probably from destroying the robot in the middle of transporting a crop). So instead of binding to a crop, the transvector dislocator binds to itself and teleports to the spot in front of it.

SOLUTION: Try breaking both the robot and the transvector dislocator to reset the natural order of the transvector binder. To avoid this issue in the future, do not break the robot while it is actively transporting a crop.

**Reply by Grubasov** [reply]
Thanks for help but that wasn't a problem. Water source on fields was causing this issue. Garden soil is must have in this farm.

**Reply by DylanTaylor1** [reply]
@Grubasov Not True! I actually see the issue from your first screenshot above.

ISSUE: The farm is not compatible with water sources.

WHY THIS HAPPENS: There MUST be a block on top of the water source such as a lilypad or glowstone. Does not matter what it is so long as there is a block. Without one, the robot will try to place a cropstick on the water (which it cannot) and, if transporting a crop, try to click it with the transvector binder (which it cannot) so the natural order of the binder is once again messed up.

SOLUTION: Place a block of some sort over your water source.

**Reply by Grubasov** [reply]
Ok thank you very much for help

**Reply by rob-people** [reply]
I'm having the same issue, but water sources are not the issue. Instead it seems to be related to increasing the storage area or work area, I haven't narrowed down exactly which yet, or determined if it's both.

Farm layout below:

![2025-06-07_21 36 04](https://github.com/user-attachments/assets/13226383-4680-4a40-906b-9e19fe346a68)

The screenshot is of the farm working perfectly with default size settings, previously when I was running it with settings that matched the actual size of the working and storage plots (8x8, 16x16) the robot would function correctly while filling the first column of storage farm, then at some point during the second column would incorrectly move the Dislocator onto the farmland that connects the two farms, after which it would of course no longer be able to move any crops. 

I tried the solution from the first reply here (break the robot and dislocator to reset their states) several times, and eventually tried restoring settings to default to see if that was the problem, and the problem went away.

The only sections of config.lua which were changed from default were to change this section:

```
-- Side Length of Working Farm
workingFarmSize = 6,
-- Side Length of Storage Farm
storageFarmSize = 9,
```

To this:

```
-- Side Length of Working Farm
workingFarmSize = 8,
-- Side Length of Storage Farm
storageFarmSize = 16,
```

**Reply by DylanTaylor1** [reply]
I see two potential problem areas.
1) "The top row of the working farm will always align with the top row of the storage farm" but it looks like your storage farm is two blocks shorter.
2) "The transvector dislocator also has a maximum range of 16 blocks." That is radial from the transvector dislocator so eventually it reaches a column on the storage farm that's just too far if using a 16x16 plot. It's been a while but I think something like 13x13 is as big as you want to go.

**Reply by rob-people** [reply]
I had missed the text that the top row of the working farm and storage farm must align, that conflicts with the visual guide in which the working farm expands upwards and left while the storage farm expands downwards and right, but it would certainly explain the behaviour I'm seeing. 

I was aware of the transvector binder/dislocator range limits, with my layout above I was able to manually test moving crops to/from the furthest corners of my storage farm with no issues, but that will certainly change if the top row of the storage farm must align with the top row of the working farm. I'll make it small enough to fit in range after the change.

Thanks for the help!

**Reply by DylanTaylor1** [reply]
Sorry for the confusion, I really couldn't think of a good way to show that on the image.

**Comment by TacticalUnicorn** [comment]
Hi some help please. I hope this is not a silly mistake. I began the farm in a checkerboard pattern of oak bonsai trees. I am running autostat and autospread. The robot is destroying all the cross breeded crops and a few of the originals. It is destroying every seed. What silly mistake have i made?? I checked the common errors and it was none of those
![2024-06-24_21 19 03](https://github.com/DylanTaylor1/GTNH-CropAutomation/assets/132719973/18d06120-9e8a-415e-8eb8-037d0b86884e)

**Reply by TacticalUnicorn** [reply]
To elaborate further, when a new crop is bred, the robot tries to transvector dislocator it, but the crop stick and crop both drop as items

**Reply by TacticalUnicorn** [reply]
Found the solution. I had misread and place the transvector locator facing north when it should of been south. Thanks.

**Comment by bunducdragos** [comment]
Hello,

There is a problem with the working area for the crops.

Let's say you have 2 parents 21 31 0 stats for both, the child that will grow will have 23 31 2 for example, it will be unable to live and will die out and dissapear from the crop sticks. The program is leaving the crop sticks like this without making them into cross breeding ones. The robot will be left with just clearing out weeds.
![Screenshot 2024-06-26 191554](https://github.com/DylanTaylor1/GTNH-CropAutomation/assets/22751148/7abdc847-b2dc-4221-a70f-b1a41725bebe)

**Reply by DylanTaylor1** [reply]
Hello,

I just looked into a possible safe-guard against this happening and unfortunately it is impossible to tell the difference between a single and double cropstick with the component.geolyzer. Therefore, the robot can only work with what it _expects_ to be on the farms. If some random event happens such as a crop dying out (from pollution or poor environment stats or player interference) then it won't actually know about that. However, If a weed grows on these single cropsticks then the protocol is to deweed and place a cropstick to turn it back into a double cropstick.

**Comment by issaOni** [comment]
Hello,

When I try to wget the code the download fails or it says the file already exists but doesn't have an option to install is there any way that I could fix this.
![Screenshot 2024-07-11 140431](https://github.com/DylanTaylor1/GTNH-CropAutomation/assets/175347609/eb33d890-6290-4170-8e39-74b48e788317)
![Screenshot 2024-07-11 141125](https://github.com/DylanTaylor1/GTNH-CropAutomation/assets/175347609/56d6d43d-8135-474e-84d3-bccf66cde65d)

**Reply by DylanTaylor1** [reply]
Hey there!

The "PKIX Path Building" Error is covered under the Troubleshooting section, but essentially you need to update your java to get the certificates necessary to download files from GitHub.

Upon failing to download, it must have created an empty file "setup.lua" which is why it says the file already exists. You can type "ls" to see the files installed on the robot and "rm setup.lua" to delete the empty file.

Also typing ./install <filename> is not how you install scripts. What's actually happening is the robot is trying to run a file called "install" which does not exist, hence the file not found error.

Hope that helps!

**Reply by Browniehound** [reply]
I'm having this same issue, I have uninstalled java and installed the newest x64 version, I have restarted my computer multiple times, and I have deleted and tried to install code multiple times, yet I still keep getting pic related. Idk if it matters but I have starlink for internet and I'm playing on a single player world on my pc.
![2025-01-11_21 12 42](https://github.com/user-attachments/assets/50ff1652-6c9c-4339-b744-602a42c24408)

**Reply by DylanTaylor1** [reply]
Do you have prism? It may be necessary to specify the java version in the launcher settings. Not sure if other clients allow you to do that.

**Reply by Browniehound** [reply]
Im using the technic launcher, lemme see if I can specify it

**Reply by Browniehound** [reply]
You can specify the version but I only have the latest version installed
![Screenshot (36)](https://github.com/user-attachments/assets/1ad6c5af-4ecb-444a-8001-6cd040ba1df1)

**Comment by ArthurLenzi** [comment]
Hello! My robot stopped working after some hours. It stays above the charger but the energy only decreases (The charger is on and 100% speed). I am trying to start the program but nothing happens. I tried to throw the error on a log file but it remains empty.

Do you know what might be happening?

**Reply by DylanTaylor1** [reply]
Hello!

The only thing I can think of is make sure the charger is actually on. Perhaps even try breaking and replacing the robot. It won't move anywhere until it is fully charged. Wish I could offer more help, but I have not heard of this issue before.

**Comment by thejims** [comment]
Using autoSpread on a crop field with crops that have already crossbread and match criteria to be stored hits errors in spreadOnce:111 -> checkChild:40 -> findEmpty:17 during the first run due to the farm database not being fully populated yet.

To fix this I edited autoSpread.lua and added a continue label after the checkChild/checkParent checks (https://github.com/DylanTaylor1/GTNH-CropAutomation/blob/main/autoSpread.lua#L115) and goto continue at the end of the firstRun check statement (before https://github.com/DylanTaylor1/GTNH-CropAutomation/blob/main/autoSpread.lua#L108)

**Reply by DylanTaylor1** [reply]
This is really helpful, thank you. The firstRun check is rather new and perhaps needed some more testing. I will fix this ASAP.
